packages/core/src/compiler.ts

```diff
@@ -182,16 +187,20 @@ export default class Compiler {
           }
         }
 
-        args.push(JSON.stringify(this.plural.locale));
+        if (!isLocaleSpecificFormattingModule) {
+          args.push(JSON.stringify(this.plural.locale));
+        }
```

Comment:
This change needs to be reverted. We still want the locale as an argument because we should not assume that each locale will have an independent implementation. It should still be possible to e.g. say that all English (en, en-US, en-CA, en-UK, ...) use the same implementation, but possibly with some region-specific considerations.

Reply:
5bf4e84